### DISCUSSION TITLE
Updating the codebase with stronger typesafety

### DISCUSSION THREAD

**Discussion by rowingdude** [bxparks/AUnit#124] [General]
@bxparks 

I came across your repo while looking for Arduino test suites to incorporate into my CI/CD for AVR development. I saw a number of your contributions were over a year old and on a quick reading of a few files, I saw we can improve a few things. 

I'd like to submit a few PRs, but my work-flow is to use a discrete commit system (pretty much one or two changes per commit) for better tracking of changes.

How should I proceed? I'd hate to flood you with requests.

Thanks in advance and I look forward to hopefully contributing to your project.

**Comment by bxparks** [comment]
Can you describe these improvements before sending me PRs?

I'm busy with a bunch of other projects (programming or otherwise), and I don't have much time for Arduino stuff right now. AUnit is essentially "done" in my mind, but I can take a look what you would like to improve.

Also, converting to a Discussion, since this is an open-ended discussion, instead of a specific bug or feature request.
